### ISSUE TITLE
Pretzel plugins list / distribution

### ISSUE THREAD

**Issue by laedit** [Code52/pretzel#226]
So Pretzel has the ability to use plugins now, and people ultimately will start to creates some, like @AndreyAkinshin for [Categories and Tags pages](https://github.com/Code52/pretzel/issues/208).

In order to share those plugins I think we need at least to list the known plugins on a wiki page, and maybe provide some way to download a plugin and install it in the right folder through a pretzel command.

@shiftkey, @JakeGinnivan what are your thoughts about this?

**Comment by AndreyAkinshin** [comment]
@laedit, it is a great idea, I like it.

**Comment by JakeGinnivan** [comment]
Sounds good. Maybe via Nuget or something?

Also, is it worth looking at #217. That could make writing plugins much more lightweight

**Comment by laedit** [comment]
Yeah, but I think that both plugin type (dll or scriptcs) needs an easy way to dowload/install.
Nuget can do the trick, I need to study that, thanks for the idea :)

**Comment by thoemmi** [comment]
I'd like to write an plugin or two (e.g. porting [redirect_from](https://github.com/jekyll/jekyll-redirect-from)). I can't find a *Pretzel* (or *Pretzel.Logic*) NuGet package. How are potential plugin authors expected to write a plugin?

**Comment by laedit** [comment]
Pretzel is available only through [Chocolatey](https://chocolatey.org/packages/Pretzel) or [GitHub Releases](https://github.com/Code52/pretzel/releases).
You only need to reference the exe directly from your plugin project.
But when I'll have the time to release the next version of Pretzel you'll have the possibility to create a plugin with scriptcs and in that case there is no need to reference the exe.
I plan to write some doc about the extensibility but I haven't the time lately.

**Comment by thoemmi** [comment]
I wrote two "classic" plugins implementing the `ITransform` interface, [Pretzel.RedirectFrom](https://github.com/thoemmi/Pretzel.RedirectFrom) and [Pretzel.Sitemap](https://github.com/thoemmi/Pretzel.Sitemap), and I had to add `pretzel.exe` to the repositories instead of referencing a nuget package. :confused: 

I wasn't able to setup working unit tests for my projects because several assemblies are merged into Pretzel. Maybe Scriptcs is a better way for writing plugins. I'll have a look at the sources to understand it, otherwise I'll wait for your documentation :wink:

**Comment by laedit** [comment]
It's not ideal but I don't know it it's worth it to make a Nuget package only for plugins.
But yeah, ScriptCs is a very cool way for writing plugins :smiley:
